department-of-veterans-affairs/caseflow
# Add cavc transcription admin action

Bumps [CASEFLOW-107]

### Description
Allows cavc users to add transcription admin actions.

### Acceptance Criteria
- [ ] Task action "Send to transcription team" is added to SendCavcRemandProcessedLetterTask
- [ ] Provide Instructions and context for this action appears on the submit admin actions page
- [ ] Creates a TranscriptionTask assigned to the TranscriptionTeam as a child of the distribution task

### Testing Plan
1. Log in as the CAVC team admin and assign any open `SendCavcRemandProcessedLetterTask` to yourself
1. Click the task action dropdown and confirm new "send to transcription team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a transcription task assigned to the trascription team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the trancription team
```ruby
Appeal 928 (evidence_submission)                       ID   STATUS      ASGN_BY                ASGN_TO                UPDATED_AT
└── RootTask                                           2781 on_hold                            Bva                    2020-11-16 15:55:54 UTC
    └── DistributionTask                               2782 on_hold                            Bva                    2020-11-16 15:55:54 UTC
        ├── CavcTask                                   2783 on_hold                            Bva                    2020-11-16 15:55:54 UTC
        │   └── SendCavcRemandProcessedLetterTask      2784 on_hold                            CavcLitigationSupport  2020-11-16 20:52:28 UTC
        │       └── SendCavcRemandProcessedLetterTask  3363 in_progress CAVC_LIT_SUPPORT_ADMIN CAVC_LIT_SUPPORT_ADMIN 2020-11-16 20:52:34 UTC
        └── TranscriptionTask                          3364 assigned    CAVC_LIT_SUPPORT_ADMIN TranscriptionTeam      2020-11-16 20:53:15 UTC
```

### User Facing Changes

 Workflow Step|UI
 ---|---
New task action|<img width="982" alt="Screen Shot 2020-11-16 at 4 17 56 PM" src="https://user-images.githubusercontent.com/45575454/99309382-5338d080-2827-11eb-8f0e-5458fb8bc141.png">
Modal validation|<img width="462" alt="Screen Shot 2020-11-16 at 3 53 09 PM" src="https://user-images.githubusercontent.com/45575454/99308923-aa8a7100-2826-11eb-9eba-060031e2faf9.png">
Success & updated task snapshot|<img width="736" alt="Screen Shot 2020-11-16 at 3 53 19 PM" src="https://user-images.githubusercontent.com/45575454/99308928-aceccb00-2826-11eb-9651-60b3e17816bf.png">
Error creating duplicate task|<img width="469" alt="Screen Shot 2020-11-16 at 3 53 28 PM" src="https://user-images.githubusercontent.com/45575454/99308940-b0805200-2826-11eb-93b4-fd5de2c55258.png">



### Code Documentation Updates
